tests/pathfinding/test_capacity_updates.py

```diff
@@ -0,0 +1,409 @@
+        token_network.address] = token_network
+
+    token_network.handle_channel_opened_event(
+        channel_identifier=ChannelIdentifier(0),
+        participant1=addresses[0],
+        participant2=addresses[1],
+        settle_timeout=15,
+    )
+
+    token_network.handle_channel_new_deposit_event(
+        channel_identifier=ChannelIdentifier(0),
+        receiver=addresses[0],
+        total_deposit=100,
+    )
+
+    token_network.handle_channel_new_deposit_event(
+        channel_identifier=ChannelIdentifier(0),
+        receiver=addresses[1],
+        total_deposit=100,
+    )
+
+    # Check that the new channel has id == 0
+    assert token_network.channel_id_to_addresses[ChannelIdentifier(0)] == (
+        addresses[0],
+        addresses[1],
+    )
+
+    message = get_UpdatePFS_message(
+        updating_participant=addresses[2],
+        other_participant=addresses[1],
+    )
+
+    with pytest.raises(InvalidCapacityUpdate) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message)
+    print(str(exinfo.value))
+    assert 'Sender of Capacity Update does not match' in str(exinfo.value)
+
+
+def test_pfs_rejects_capacity_update_with_wrong_other_participant(
+        addresses: List[Address],
+        pathfinding_service_mocked_listeners: PathfindingService,
+):
+    pathfinding_service_mocked_listeners.chain_id = 1
+
+    token_network = TokenNetwork(
+        token_network_address=DEFAULT_TOKEN_NETWORK_ADDRESS,
+        token_address=DEFAULT_TOKEN_ADDRESS,
+    )
+
+    pathfinding_service_mocked_listeners.token_networks[
+        token_network.address] = token_network
+
+    token_network.handle_channel_opened_event(
+        channel_identifier=ChannelIdentifier(0),
+        participant1=addresses[0],
+        participant2=addresses[1],
+        settle_timeout=15,
+    )
+
+    token_network.handle_channel_new_deposit_event(
+        channel_identifier=ChannelIdentifier(0),
+        receiver=addresses[0],
+        total_deposit=100,
+    )
+
+    token_network.handle_channel_new_deposit_event(
+        channel_identifier=ChannelIdentifier(0),
+        receiver=addresses[1],
+        total_deposit=100,
+    )
+
+    # Check that the new channel has id == 0
+    assert token_network.channel_id_to_addresses[ChannelIdentifier(0)] == (
+        addresses[0],
+        addresses[1],
+    )
+
+    message = get_UpdatePFS_message(
+        updating_participant=addresses[0],
+        other_participant=addresses[2],
+    )
+
+    with pytest.raises(InvalidCapacityUpdate) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message)
+    print(str(exinfo.value))
+    assert 'Other Participant of Capacity Update does not match' in str(exinfo.value)
+
+
+def test_pfs_rejects_capacity_update_with_wrong_nonces(
+        addresses: List[Address],
+        pathfinding_service_mocked_listeners: PathfindingService,
+):
+    pathfinding_service_mocked_listeners.chain_id = 1
```

Comment:
So looking at all these tests, there is a **lot of** duplicated code. Can't you abstract away the setup of the tests into a common function and then use it in all of the tests?

Reply:
This is a good point. I do not have much time today refactoring this. I will open an issue and get this done next week. Would that be sufficient?

Reply:
Sure, as long as we have a task for it this is fine.

Reply:
ok

Reply:
#200